src/app/search/search.component.ts

```diff
@@ -151,82 +153,69 @@ export class SearchComponent implements OnInit, OnDestroy {
         distinctUntilChanged(),
         takeUntil(this.ngUnsubscribe)
       )
-      .subscribe((value: string) => {
-        this.values = value;
-        this.onKey();
+      .subscribe((searchText: string) => {
+        this.onKey(searchText);
       });
     this.hits = [];
-    this.advancedSearchObject = {
-      ANDSplitFilter: '',
-      ANDNoSplitFilter: '',
-      ORFilter: '',
-      NOTFilter: '',
-      searchMode: 'files',
-      toAdvanceSearch: false
-    };
-    this.parseParams();
 
-    this.aNDSplitFilterText$ = this.advancedSearchService.advancedSearch$.pipe(
-      map((advancedSearchObject: AdvancedSearchObject) => this.searchService.joinComma(advancedSearchObject.ANDSplitFilter))
-    );
-    this.aNDNoSplitFilterText$ = this.advancedSearchService.advancedSearch$.pipe(
-      map((advancedSearchObject: AdvancedSearchObject) => this.searchService.joinComma(advancedSearchObject.ANDNoSplitFilter))
-    );
-    this.oRFilterText$ = this.advancedSearchService.advancedSearch$.pipe(
-      map((advancedSearchObject: AdvancedSearchObject) => this.searchService.joinComma(advancedSearchObject.ORFilter))
-    );
-    this.nOTFilterText$ = this.advancedSearchService.advancedSearch$.pipe(
-      map((advancedSearchObject: AdvancedSearchObject) => this.searchService.joinComma(advancedSearchObject.NOTFilter))
-    );
-    this.advancedSearchService.advancedSearch$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((advancedSearch: AdvancedSearchObject) => {
-      this.advancedSearchObject = advancedSearch;
-      // Upon init, the user did not want to do an advanced search, but this triggers anyways.  Using toAdvanceSearch to stop it.
-      if (advancedSearch.toAdvanceSearch) {
-        this.updateQuery();
-      }
+    this.aNDSplitFilterText$ = this.advancedSearchQuery.aNDSplitFilterText$;
+    this.aNDNoSplitFilterText$ = this.advancedSearchQuery.aNDNoSplitFilterText$;
+    this.oRFilterText$ = this.advancedSearchQuery.oRFilterText$;
+    this.nOTFilterText$ = this.advancedSearchQuery.nOTFilterText$;
+    // The reason why we have this here is because the updatePermalink function isn't in a service...
+    // because the function modifies something that's in this component and not in the state.
+    // TODO:move it to the state
+    this.advancedSearchQuery.advancedSearch$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(() => {
+      this.updatePermalink();
     });
   }
 
   /**
    * Applies parameters from the permalink to the search
    */
   parseParams() {
-    let useAdvSearch = false;
     const paramMap: ParamMap = this.activatedRoute.snapshot.queryParamMap;
     if (!paramMap) {
       return;
     }
+    const newAdvancedSearchObject: AdvancedSearchObject = initialAdvancedSearchObject;
+    let newFilters: Map<string, Set<string>> = new Map<string, Set<string>>();
+    // URL is gospel, if it doesn't have a search term, then there's no search term
+    if (!paramMap.has('search')) {
```

Comment:
'search' could be a constant; it's used 5 times. In general it would be nice if there were an interface with all the difference queryParams we honor.